aderyn_core/src/detect/low/constant_funcs_assembly.rs

```diff
@@ -0,0 +1,131 @@
+use std::collections::BTreeMap;
+use std::error::Error;
+
+use crate::ast::{ASTNode, NodeID, StateMutability};
+
+use crate::capture;
+use crate::context::browser::ExtractInlineAssemblys;
+use crate::context::investigator::{
+    StandardInvestigationStyle, StandardInvestigator, StandardInvestigatorVisitor,
+};
+use crate::detect::detector::IssueDetectorNamePool;
+use crate::detect::helpers;
+use crate::{
+    context::workspace_context::WorkspaceContext,
+    detect::detector::{IssueDetector, IssueSeverity},
+};
+use eyre::Result;
+
+#[derive(Default)]
+pub struct ConstantFunctionContainsAssemblyDetector {
+    // Keys are: [0] source file name, [1] line number, [2] character location of node.
+    // Do not add items manually, use `capture!` to add nodes to this BTreeMap.
+    found_instances: BTreeMap<(String, usize, String), NodeID>,
+}
+
+impl IssueDetector for ConstantFunctionContainsAssemblyDetector {
+    fn detect(&mut self, context: &WorkspaceContext) -> Result<bool, Box<dyn Error>> {
+        for function in helpers::get_implemented_external_and_public_functions(context) {
+            if function.state_mutability() == &StateMutability::View
```

Comment:
This issue only presents itself if the pure/view function that contains the assembly can be compiled with solce <0.5. We should filter for this also, otherwise we'll get lots of false positives for valid assembly within pure functions in 0.8.0, for example.

Reply:
✅